src/main/java/hudson/tasks/Ant.java

```diff
@@ -341,7 +354,8 @@ public void buildEnvVars(EnvVars env) {
          * Gets the executable path of this Ant on the given target system.
          */
         public String getExecutable(Launcher launcher) throws IOException, InterruptedException {
-            return launcher.getChannel().call(new Callable<String,IOException>() {
+            return launcher.getChannel().call(new MasterToSlaveCallable<String,IOException>() {
+                private static final long serialVersionUID = 906341330603832653L;
```

Comment:
> why is it safer to have it in a `static` nested class?

Different Java compilers are allowed to produce slightly different signatures of anonymous inner classes, which can affect the serialized form. In general serializable classes should be top-level or `static` to avoid synthetic `this$0` members etc.